tests/integration/integration-test.py

```diff
@@ -62,6 +64,188 @@ def test_get_notes_path(self, endpoint='/notes'):
             schema = utils.get_resource_schema(self, 'Error')
             utils.check_schema(self, response, schema)
 
+    def get_response(self, params, endpoint='/notes'):
+        response = utils.make_request(self, endpoint, 200, params=params)
+        schema = utils.get_resource_schema(self, 'NoteResource')
+        utils.check_schema(self, response, schema)
+        return response
+
+    # Test case: GET /notes?studentId
+    def test_get_notes_query_by_student_id(self, endpoint='/notes'):
+        for student_id in self.test_cases['valid_student_id']:
+            params = {'studentId': student_id}
+            response = self.get_response(params)
+
+            creator_ids = []
+            context_types = []
+            notes = []
+            sources = []
+            # Validating the studentId requested is the same studentId received
+            for resource in response.json()['data']:
+                attributes = resource['attributes']
+                actual_student_id = attributes['studentId']
+                self.assertEqual(actual_student_id, student_id)
+
+                note_id = resource['id']
+                match_student_id = re.match(r'(^\d{9})(\b-)', note_id)
+                if match_student_id:
+                    self.assertEqual(student_id, match_student_id.group(1))
+                else:
+                    self.fail('Note id don\'t include a student id number')
+
+                creator_ids.append(attributes['creatorId'])
+                context_types.append(attributes['context']['contextType'])
+                notes.append(attributes['note'])
+                sources.append(attributes['source'])
+
+            self.query_creator_id(student_id, creator_ids)
+            self.query_string_search(student_id, notes)
+            self.query_sort_keys(student_id)
+            self.query_source(student_id, sources)
+            self.query_context_type(student_id, context_types)
```

Comment:
Instead of fetching these testing information/cases whenever we run the integration test, I think a better way to prevent making too many requests is to define some test cases in your config file and don't let these tests depend on each other. There are a couple of main reasons why I think we should implement in that way:

1. In your current method, it's very possible that some students only have one or none note in their folder. For those who only have less than one note, testing with the sorting query doesn't make any sense. Similarly, if a student only has one context type of notes, testing it with `contextType` filters also isn't that valuable.
2. Currently, no matter what, for each student ID will always run into these sub test cases. It's pretty hard to tell if the error happens and, in fact, you repeat so many similar tests cases. For example, if `student 1` has notes with `creatorId: 1234` and `student 2` also has notes with the same `creatorId`, they both will test `self.query_creator_id(student_id, '1234')`. Although the `student_id` could be different, the logic you were testing is exactly the same.

What do you think?